Automattic/wp-calypso
# BBE: Save website content to server

#### Proposed Changes

UI Changes:
* Adds a new button on the website content form to save website content to the server. Clicking on the button will make a network request that saves all form data to the server.
<img width="925" alt="image" src="https://user-images.githubusercontent.com/5436027/212608713-633ce2c0-91ab-4e9f-b45f-235748ac7401.png">

* The "Save Changes" button is enabled only if there are unsaved changes.
* The "Next" button also saves content first and then opens the next section.
* Clicking on any of the section titles opens the relevant section and saves content in the background.
* For the above two actions, the network request to save changes is only sent if there are unsaved changes.
* Refreshing the page or opening the page in a different browser/tab/device should pre-populate the form with the saved data.

Code changes:
* While the quantity of changes seems to be large, they are mostly due to refactors, considering that the source of truth for this data is the server and not browser-based storage. I have left inline comments to explain major changes.

#### Testing Instructions



* Go to `/start/do-it-for-me`, proceed through the flow steps and complete the purchase.
* Once you reach the website content form, type in any text content and click on the "Save Changes" button.
* Confirm that the button has a loading state and form interaction is blocked during this time.
* Reload the page and confirm that the typed data is prepopulated.
* Repeat the above three steps with the site logo section and the feedback section. 
* Open the form link in an incognito window or in a different browser and confirm that the form is prepopulated with saved data.

#### Pre-merge Checklist





Fixes Automattic/martech#1405
